### PR TITLE
Email line manager on new annual leave request

### PR DESCRIPTION
**Pull request by johnny-m-young**
# What
- Sets up GOV.UK Notify within the application
- Creates an EmailsHelper which is used to send emails via GOV.UK Notify
- Uses the EmailsHelper to send user's line manage an email when they successfully raise an annual leave request (on the `annual_leave_requests#create` action)

# Why
This eliminates the manual process of letting your line manager know when you have requested annual leave.

### CODE REVIEW

File: app/helpers/emails_helper.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module EmailsHelper
```
Comment (ryanb-gds) [inline app/helpers/emails_helper.rb:1]: I'm not strongly opposed to it, but I don't know if this is a perfect fit for a helper. I usually consider helpers most appropriate for creating abstractions that can be reused by multiple controllers, especially for use with views. In this situation I would probably create an `as_manager_email_hash` method on the `AnnualLeaveRequest` model to do the serialization to the email hash and use the notify client directly in the controller. The serialization is so tightly coupled to the model that it doesn't make sense to have it as a separate module to my mind.

File: spec/controllers/annual_leave_requests_controller_spec.rb [hunk 2]
```diff
@@ -1,8 +1,12 @@
 RSpec.describe AnnualLeaveRequestsController do
   describe "POST create" do
-    let(:user) { create(:user) }
+    let(:user) { create(:user, line_manager_id: line_manager.id) }
+    let(:line_manager) { create(:user, email: "line_manager@digital.cabinet-office.gov.uk") }
 
     setup do
+      notify_test_client = Notifications::Client.new(ENV["NOTIFY_TEST_API_KEY"])
+      allow(controller.helpers).to receive(:client).and_return(notify_test_client)
+
```
Comment (ryanb-gds) [inline spec/controllers/annual_leave_requests_controller_spec.rb]: I'm not convinced that this actually confirms that an email would have been sent, and it certainly doesn't check the expected parameters were provided. You could argue that the client wouldn't have been instantiated if an email hadn't been sent, but it's a bit indirect. You could use an RSpec spy to check the client was asked to send the email with the expected hash.

**Review by ryanb-gds — COMMENTED** [review]
Just an observation about testing and a question about helper usage, but looking good. Sorry for the delay from when I first started looking at it.

File: spec/controllers/annual_leave_requests_controller_spec.rb (hunk 2 above)
Comment (johnny-m-young) [reply spec/controllers/annual_leave_requests_controller_spec.rb]: Thanks Ryan, as discussed in our call I've updated this to check the controller makes the correct call to the Notify client with the correct arguments.

File: app/helpers/emails_helper.rb (hunk 1 above)
Comment (johnny-m-young) [reply app/helpers/emails_helper.rb:1]: As discussed in our call, in the interest of time I am going to leave this as a helper.

**Pull request merged**